File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Types/Generics.vb
Imports System.Collections

#Region "Class"

Namespace Class
    Public Class NoGenericType
    End Class

    Public Class SingleGenericType(Of T)
    End Class

    Public Class MultipleGenericType(Of T As Structure, Y As { IEnumerable, Class, New })
    End Class
End Namespace

#End Region

#Region "Struct"

Namespace Struct
    Public Structure NoGenericType
    End Structure

    Public Structure SingleGenericType(Of T)
    End Structure

    Public Structure MultipleGenericType(Of T As Structure, Y As { IEnumerable, Class, New })
    End Structure
End Namespace

#End Region

#Region "Interface"

Namespace Interface
    Public Interface NoGenericType
    End Interface

    Public Interface SingleGenericType(Of T)
    End Interface

    Public Interface MultipleGenericType(Of T As Structure, Y As { IEnumerable, Class, New })
    End Interface

    Public Interface CovariantGenericType(Of In T)
    End Interface

    Public Interface ContravariantGenericType(Of Out T)
    End Interface
End Namespace

#End Region

#Region "Delegate"

Namespace Delegate
    Public Delegate Sub NoGenericType()

    Public Delegate Sub SingleGenericType(Of T)()

    Public Delegate Sub MultipleGenericType(Of T As Structure, Y As { IEnumerable, Class, New })()

    Public Delegate Sub CovariantGenericType(Of In T)()

    Public Delegate Sub ContravariantGenericType(Of Out T)()
End Namespace

#End Region

File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Types/Generics.cs
using System.Collections;

#region Class

namespace Class
{
    public class NoGenericType
    {
    }

    public class SingleGenericType<T>
    {
    }

    public class MultipleGenericType<T, Y>
        where T : struct
        where Y : IEnumerable, class, new()
    {
    }
}

#endregion

#region Struct

namespace Struct
{
    public struct NoGenericType
    {
    }

    public struct SingleGenericType<T>
    {
    }

    public struct MultipleGenericType<T, Y>
        where T : struct
        where Y : IEnumerable, class, new()
    {
    }
}

#endregion

#region Interface

namespace Interface
{
    public interface NoGenericType
    {
    }

    public interface SingleGenericType<T>
    {
    }

    public interface MultipleGenericType<T, Y>
        where T : struct
        where Y : IEnumerable, class, new()
    {
    }

    public interface CovariantGenericType<in T>
    {
    }

    public interface ContravariantGenericType<out T>
    {
    }
}

#endregion

#region Delegate

namespace Delegate
{
    public delegate void NoGenericType();

    public delegate void SingleGenericType<T>();

    public delegate void MultipleGenericType<T, Y>()
        where T : struct
        where Y : IEnumerable, class, new();

    public delegate void CovariantGenericType<in T>();

    public delegate void ContravariantGenericType<out T>();
}

#endregion